### PR TITLE
Allow optional mounting of GPUs for PYML

### PR DESCRIPTION
**Pull request by anr-tonic**
Relevant Allos PR: https://github.com/TonicAI/allos/pull/6430

### CODE REVIEW

**Comment by akamor** [comment]
what does amount: 2 imply?

**Comment by gandersteele** [comment]
Note: PyML currently only supports acceleration with NVIDIA GPUs via CUDA.

File: values.sample.yaml [hunk 1]
```diff
@@ -10,6 +10,19 @@ useUnprivilegedContainers: false
 # filesystem and provides necessary emptyDir volume mounts
 readOnlyRootFilesystem: false
 
+# To enable gpu acceleration for the pyml container, uncomment the following
+# lines and provide the gpu vendor and how much to allocate to the pod. For
```
Comment (gandersteele) [inline values.sample.yaml:14]:
```suggestion
# To enable Nvidia gpu acceleration for the pyml container, uncomment the following
# lines and specify how much to allocate to the pod. For
```

**Comment by gandersteele** [comment]
How to test? In the past, we've started a training job, ssh'd into vm and ran `watch nvidia-smi` to make sure the gpu is being used. If helpful, I can make a branch of pyml which logs this more directly

**Comment by anr-tonic** [comment]
@akamor The number is the amount of cards. I lowered the example to 1.

@gandersteele @calexander3 and I can look into getting a GPU node set up, but as far as I can tell this is all that's required on our end. I hardcoded the requested resource to `nvidia.com/gpu` which is what the nvidia kubernetes plugin exposes as a resource.

**Comment by gandersteele** [comment]
> @gandersteele @calexander3 and I can look into getting a GPU node set up, but as far as I can tell this is all that's required on our end. I'm also gonna look to see if it is possible to test this locally via minikube or something. I hardcoded the requested resource to `nvidia.com/gpu` which is what the nvidia kubernetes plugin exposes as a resource.

That would be great! NB I think there is some pain around installing the appropriate [nvidia drivers](https://docs.nvidia.com/datacenter/cloud-native/kubernetes/install-k8s.html#install-nvidia-drivers). I dont know how yall deploy your nodes, but the aws [deep learning AMIs](https://docs.aws.amazon.com/dlami/latest/devguide/overview-base.html) have all the appropriate drivers pre-installed.

**Comment by anr-tonic** [comment]
I have a GPU enabled node in the tonic eks cluster. It only has 1 card to keep costs down, but I've been able to access the card from both pods and containers on the host. However @gandersteele discovered that PYML isn't able to access the card and as far as I know is still investigating what is happening there; however, @gandersteele lemme know if that's wrong and/or you need more assistance from me on poking and prodding things.

After the pyml gpu issue is resolved we should be good to merge this PR.

**Review by gandersteele — APPROVED** [review]
LGTM

**Pull request merged**